django/db/views.py

```diff
@@ -1635,7 +1644,9 @@ def __init__(self,dataset,is_authenticated=False):
             self.has_otherreagents_for_datarecords(self.dataset_id)
         self.small_molecule_queryset = \
             self.small_molecules_for_dataset(self.dataset_id)
-                
+        self.has_small_molecules_for_datarecords = \
```

Comment:
Shadowing the method with an attribute like this is a bad idea. The method or the attribute should be renamed. Also, why is self.dataset_id being passed into the method -- can't the method just look up self.dataset_id itself?